src/plugin.validation.dns.cloudflare/Cloudflare.cs

```diff
@@ -85,13 +85,10 @@ private async Task DeleteRecord(string recordName, string token, IAuthorizedSynt
                 .Match(MatchType.All)
                 .CallAsync(_hc)
                 .ConfigureAwait(false);
-            var record = records.FirstOrDefault();
-            if (record != null)
-            {
-                await dns.Delete(record.Id)
-                    .CallAsync(_hc)
-                    .ConfigureAwait(false);
-            }
+            var record = records.First();
+            await dns.Delete(record.Id)
+                .CallAsync(_hc)
+                .ConfigureAwait(false);
```

Comment:
I don't see the logic of this change. Why error out when the record doesn't exist? The `.First()` would throw a vague index out of range exception probably. If you want to warn about a record that was expected to exist but didn't (as indication of a possible bug?), we should throw an exception specifically stating that.

Reply:
You're right - changed it accordingly.